src/components/Editor.js

```diff
@@ -5,11 +5,18 @@ import 'brace/theme/github';
 import 'brace/ext/language_tools';
 import customCompleter from './customCompleter.js'
 
+import 'brace/ext/searchbox';
+
 /**
 * @summary - Editor is a React Component that creat the Ace Editor in the DOM.
 *
 */
+
 class Editor extends Component {
+  componentWillUnmount() {
+    //Updates state in reducer before closing editor
+    this.props.refresh(window.ace.edit("ace-editor").getSession().getValue(), this.props.user ? this.props.user.uid : 'anon');
```

Comment:
I'm testing this now. It does do what it says. However it is a little confusing as it doesn't also render. Can you add a render cycle here too?

Reply:
Should be done now